fabric/bin/peer.sh

```diff
@@ -433,24 +435,44 @@ handle_lifecycle_chaincode_createenclave() {
         ORDERER_ADDR=$2;
         shift; shift
         ;;
+        -s|--sgx-credentials-path)
+        SGX_CREDENTIALS_PATH=$2
+        ATTESTATION_PARAMS=$(jq -c -n --arg atype "$(cat ${SGX_CREDENTIALS_PATH}/spid_type.txt)" --arg spid "$(cat ${SGX_CREDENTIALS_PATH}/spid.txt)" --arg sig_rl "" '{attestation_type: $atype, hex_spid: $spid, sig_rl: $sig_rl}' | base64 --wrap=0)
+        shift; shift
+        ;;
+        --peerAddresses)
+        PEER_ADDRESS=$2
+        shift; shift
+        ;;
         *)
         die "createenclave: invalid option"
         ;;
         esac
     done
 
+    # peer address must be specified in createenclave
+    [ -z "${PEER_ADDRESS}" ] && die "No peer address specified in createenclave"
+
     # - createenclave can only be run on FPC chaincodes
     FILES_NUM=$(ls -1 ${FABRIC_STATE_DIR}/is-fpc-c-chaincode.${CC_NAME}.* 2> /dev/null | wc -l) 
     if [ ${FILES_NUM} -eq 0 ]; then
         die "createenclave: $CC_NAME is not written in language 'fpc-c'"
     fi
 
+    # embed json in protobuf message and b64 encode it
+    ATTESTATION_PARAMS_PROTO=$(echo "parameters: \"${ATTESTATION_PARAMS}\"" | protoc --encode attestation.AttestationParameters --proto_path=${FPC_TOP_DIR}/protos/fpc ${FPC_TOP_DIR}/protos/fpc/attestation.proto | base64 --wrap=0)
+
+    # create host params
+    HOST_PARAMS=$(echo "${PEER_ADDRESS}" | base64 --wrap=0)
```

Comment:
related to my comment on `fpc.proto` I suggest we handle the peer endpoint as a simple string. maybe some additional input validation if the string is in the correct format would be good (or a todo comment would also work for the moment).

Reply:
no strong opinion, done

Reply:
let's just remove the base64 conversion here

Reply:
again, no strong opinion, done